src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -166,7 +193,7 @@ public bool IsPinOpen(int pinNumber)
         /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
         /// <param name="mode">The mode to check.</param>
         /// <returns>The status if the pin supports the mode.</returns>
-        public bool IsPinModeSupported(int pinNumber, PinMode mode)
+        public virtual bool IsPinModeSupported(int pinNumber, PinMode mode)
```

Comment:
The implementation of this is using _openPins list which is private (instead of protected) as well as _driver, which is also private. Should those two fields be protected instead so that subclasses can use them in their implementation?

Reply:
I also think that we need to have access to the collection maybe via protected methods instead of giving access to the collection itself? But is we can override things like the pincount property and ispinopen there is also the need to have a way to check the collection